src/controllers/SubjectIdentitiesController.ts

```diff
@@ -0,0 +1,62 @@
+import { Post } from '@overnightjs/core';
+/*
+ * Copyright (c) 2021, IBM Deutschland GmbH
+ */
+
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+
+import { Response } from 'express';
+
+import { Controller, Get, Middleware } from '@overnightjs/core';
+import { ISecureRequest } from '@overnightjs/jwt';
+import { Logger } from '@overnightjs/logger';
+
+import { SubjectIdentitiesModel } from '../models/SubjectIdentitiesModel';
+import { AuthorizationController } from './AuthorizationController';
+import { runInNewContext } from 'node:vm';
+
+/**
+ *  Endpoint class for all subjectId management related restful methods.
+ *
+ * @export
+ * @class SubjectIdentitiesController
+ */
+@Controller('subjectIdentities')
+export class SubjectIdentitiesController {
+    private subjectIdentityModel: SubjectIdentitiesModel = new SubjectIdentitiesModel();
+
+    /**
+     * Add new subject.
+     * Is called by API user to register a new subject which will then get access to app by their id.
+     */
+     @Post('addNew')
+     @Middleware([AuthorizationController.checkApiUserLogin])
+     public async addSubjectIdentity(req: ISecureRequest, resp: Response) {
+         try {
+             // validate parameter existance
+             if (!req.body.subjectIdentity || !req.body.subjectIdentity.recordId) {
+                 return resp.status(400).send({ error: 'missing_data' });
+             }
+            
+            try {
+                var subjectIdentityExistence: boolean = await this.subjectIdentityModel.getSubjectIdentityExistence(
+                    req.body.subjectIdentity.recordId
+                );
+            } catch {//no action required as participant doesn't exist
+            }
```

Comment:
is this block really necessary? there is already a try/catch in the method called here.
respectively, i don't see why you re-throw the error caught within the method called

Reply:
Valid point, I just updated it.